pkg/codegen/schema/docs_parser.go

```diff
@@ -133,10 +133,12 @@ func (p shortcodeParser) Open(parent ast.Node, reader text.Reader, pc parser.Con
 }
 
 func (p shortcodeParser) Continue(node ast.Node, reader text.Reader, pc parser.Context) parser.State {
-	line, _ := reader.PeekLine()
+	line, seg := reader.PeekLine()
 	pos := pc.BlockOffset()
 	if pos < 0 {
 		return parser.Continue | parser.HasChildren
+	} else if pos > seg.Len() {
+		return parser.Continue | parser.HasChildren
```

Comment:
So intuitively, I wanted to return the `parser.NoChildren` state here, but I have to admit that I don't fully understand the lifecycle of these calls, so I just returned the same state as the first condition in this chain. I would have expected us to return a `parser.NoChildren` state on line 138 too and wasn't sure if that was intentional as a way to keep the parser continue to return nodes.

Reply:
I think that what you have here is correct. The content in the issue didn't look like it was malformed--do you know which line exactly this panicked on?

Reply:
> The content in the issue didn't look like it was malformed--do you know which line exactly this panicked on?

Yeah the resource that it was panicking on is the `alicloud:datahub/topic:Topic` resource. Here's the example snippet from that resource's description. Note that the example syntax still looks like HCL.

> {{% examples %}}\n## Example Usage\n\nBasic Usage\n\n- BLob Topic\n\n \nresource \"alicloud_datahub_topic\" \"example\" {\n name = \"tf_datahub_topic\"\n project_name = \"tf_datahub_project\"\n record_type = \"BLOB\"\n shard_count = 3\n life_cycle = 7\n comment = \"created by terraform\"\n}\n \n- Tuple Topic\n\n \nresource \"alicloud_datahub_topic\" \"example\" {\n name = \"tf_datahub_topic\"\n project_name = \"tf_datahub_project\"\n record_type = \"TUPLE\"\n record_schema = {\n bigint_field = \"BIGINT\"\n timestamp_field = \"TIMESTAMP\"\n string_field = \"STRING\"\n double_field = \"DOUBLE\"\n boolean_field = \"BOOLEAN\"\n }\n shard_count = 3\n life_cycle = 7\n comment = \"created by terraform\"\n}\n \n{{% /examples %}}

Reply:
Right, but do you know which line exactly it panicked on? That example is multiple lines.

Reply:
Ah yeah. I don't know that. I can find out. Regardless of that, though, shouldn't that example be omitted from the schema because it seems to have failed conversion?

Reply:
> Regardless of that, though, shouldn't that example be omitted from the schema because it seems to have failed conversion?

Yes, it should have (assuming that it did fail conversion)--I was wondering about the line so that I could better understand the fix.

Reply:
Roger that. I'll look that up and post it in the new issue that I'll open for fixing the example conversion code.

Reply:
@pgavlin I opened an issue for the issue of non-converted example being included in the schema accidentally.

As for the panic that this PR fixes, it seems to be caused when the parser encounters a whitespace character after the second new-line character after `- Tuple Topic`.

Reply:
@pgavlin if there is no other concern with the change in this PR, I'll go ahead and merge this.

Reply:
Yep, this is ready to go (I signed off)